jbpm-benchmarks/jbpm-performance-tests-jmh/src/main/java/org/jbpm/test/performance/scenario/load/services/QueryProcessesAndTasksByVariables.java

```diff
@@ -0,0 +1,249 @@
+
+    @Param("0")
+    // ! Must be overridden using -p from command line
+    private int processes;
+
+    private JBPMKieServicesController jc;
+
+    private List<Long> processIds = new ArrayList<>();
+
+    private Map<String, Object> processVariables = new HashMap<>();
+
+    private Map<String, Object> taskVariables = new HashMap<>();
+
+    private AdvanceRuntimeDataService advanceRuntimeDataService;
+    private ProcessService processService;
+    private RuntimeDataService runtimeDataService;
+    private UserTaskService userTaskService;
+    private DefinitionService definitionService;
+    private DeploymentUnit deploymentUnit;
+
+    String processDefinitionId = ProcessStorage.QueryProcessesAndTasksByVariables.getProcessDefinitionId();
+
+    @Setup
+    public void init() throws Exception {
+        log.debug("init() - starting up jBPM processes...");
+        jc = JBPMKieServicesController.getInstance(singletonList(ProcessStorage.QueryProcessesAndTasksByVariables.getPath()));
+
+        advanceRuntimeDataService = jc.getAdvanceRuntimeDataService();
+        processService = jc.getProcessService();
+        runtimeDataService = jc.getRuntimeDataService();
+        userTaskService = jc.getUserTaskService();
+        definitionService = jc.getDefinitionService();
+        deploymentUnit = jc.getDeploymentUnit();
+
+        // Get process variables
+        Map<String, String> processVarDefs = definitionService.getProcessVariables(deploymentUnit.getIdentifier(), processDefinitionId);
+        processVarDefs.entrySet().stream()
+                .filter( entry -> "integer".equalsIgnoreCase(entry.getValue()))
+                .forEach(entry -> processVariables.put(entry.getKey(), new Random().nextInt(100)));
+
+        // Start processes
+        for (int i = 0; i < processes; i++) {
+            processIds.add(processService.startProcess(deploymentUnit.getIdentifier(), processDefinitionId, processVariables));
+        }
+
+        // Get task variables
+        List<String> taskVarNames = definitionService.getTasksDefinitions(deploymentUnit.getIdentifier(), processDefinitionId).stream()
+                .flatMap( userTaskDefinition -> userTaskDefinition.getTaskInputMappings().entrySet().stream())
+                .filter(entry -> entry.getKey().contains("task_var_in") && entry.getValue().equalsIgnoreCase("Integer"))
+                .map(Map.Entry::getKey)
+                .collect(Collectors.toList());
+
+        runtimeDataService.getTasksByProcessInstanceId(processIds.get(0)).forEach( taskId -> {
+            taskVariables.putAll(userTaskService.getTaskInputContentByTaskId(taskId).entrySet().stream()
+                                         .filter( key-> taskVarNames.contains(key.getKey()))
+                                         .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue)));
+        });
+
+        // Update process variables twice
```

Comment:
I thought these task variables were already updated since they are assigned in the input mapping in the task from the process variables. But you're right, these are not being updated. My bad.